frontend/src/containers/DocList.js

```diff
@@ -13,45 +14,64 @@ import Container from '../components/Container';
 import DocumentList from '../components/app/DocumentList';
 import Overlay from '../components/app/Overlay';
 
+const EMPTY_ARRAY = [];
+const EMPTY_OBJECT = {};
+
 /**
  * @file Class based component.
  * @module DocList
  * @extends Component
  */
 class DocList extends Component {
+  shouldComponentUpdate(nextProps) {
+    const { query } = this.props;
+    const { query: nextQuery } = nextProps;
+
+    if (_.isEqual(query, nextQuery)) {
+      return false;
+    }
+
+    return true;
+  }
+
   componentDidMount = () => {
-    const { dispatch, windowType, latestNewDocument, query } = this.props;
+    const {
+      windowType,
+      latestNewDocument,
+      query,
+      getWindowBreadcrumb,
+      setLatestNewDocument,
+      selectTableItems,
+    } = this.props;
 
```

Comment:
Do you think it would be opportune in here to define a const windowId = windowType  and use it along as a prerequisite for the time when BE will provide that windowId in params instead ?

Reply:
actually you can pass the windowId from the router.js directly

Reply:
+1